remote-trait-object/src/forwarder.rs

```diff
@@ -18,43 +18,51 @@ use crate::packet::PacketView;
 use crate::port::Handler;
 use crate::service::Dispatch;
 use parking_lot::RwLock;
-use std::collections::HashMap;
+use std::collections::{HashMap, VecDeque};
 use std::fmt;
+use std::sync::Arc;
 
-type ServiceId = String;
+pub type ServiceObjectId = u32;
 
 pub struct ServiceForwarder {
-    service_handlers: RwLock<HashMap<ServiceId, Box<dyn Dispatch>>>,
+    service_objects: RwLock<HashMap<ServiceObjectId, Arc<dyn Dispatch>>>,
+    available_ids: RwLock<VecDeque<ServiceObjectId>>,
 }
 
 impl fmt::Debug for ServiceForwarder {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        f.debug_list().entries(self.service_handlers.read().keys()).finish()
+        f.debug_list().entries(self.service_objects.read().keys()).finish()
     }
 }
 
 impl ServiceForwarder {
     pub fn new() -> Self {
         Self {
-            service_handlers: Default::default(),
+            service_objects: Default::default(),
+            available_ids: RwLock::new({
+                let mut queue = VecDeque::new();
+                for i in 0..100 {
+                    queue.push_back(i)
+                }
+                queue
+            }),
         }
     }
 
-    pub fn add_service(&self, name: ServiceId, service: Box<dyn Dispatch>) {
-        let insert_result = self.service_handlers.write().insert(name.clone(), service);
-        if insert_result.is_some() {
-            panic!("Duplicated service id {}", name);
-        }
+    pub fn register_service_object(&self, service: Arc<dyn Dispatch>) -> ServiceObjectId {
+        let token = self.available_ids.write().pop_front().expect("Too many service objects had been created");
```

Comment:
IMO, we don't need use the name `token`. It does not have any meaning. 
```suggestion
        let service_object_id = self.available_ids.write().pop_front().expect("Too many service objects had been created");
```